src/main/java/dinkplugin/message/NotificationBody.java

```diff
@@ -1,39 +1,42 @@
 package dinkplugin.message;
 
+import com.google.gson.annotations.SerializedName;
 import lombok.AllArgsConstructor;
 import lombok.Builder;
-import lombok.Data;
-import lombok.EqualsAndHashCode;
-import lombok.NoArgsConstructor;
+import lombok.Singular;
 import lombok.Value;
+import lombok.With;
+import net.runelite.api.vars.AccountType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
-import java.util.LinkedList;
 import java.util.List;
 
-@Data
+@Value
+@With
 @Builder
-@NoArgsConstructor
 @AllArgsConstructor
-public class NotificationBody<T> {
-    private NotificationType type;
-    private T extra;
-    private String content;
-    private String playerName;
+public class NotificationBody<T extends Fieldable> {
 
-    @Builder.Default
-    private List<Embed> embeds = new LinkedList<>();
+    /*
+     * Dink fields
+     */
+    @NotNull
+    NotificationType type;
+    String playerName;
+    AccountType accountType;
```

Comment:
```suggestion
    transient AccountType accountType;
```

This is a new field, if we don't want it sent to the webhook server, can mark it as transient.

Reply:
I think this field is fine to pass along